adam-core/src/main/scala/org/bdgenomics/adam/rdd/feature/FeatureRDD.scala

```diff
@@ -302,35 +318,37 @@ case class DatasetBoundFeatureRDD private[rdd] (
   lazy val rdd = dataset.rdd.map(_.toAvro)
   protected lazy val optPartitionMap = None
 
-  override def saveAsParquet(filePath: String,
-                             blockSize: Int = 128 * 1024 * 1024,
-                             pageSize: Int = 1 * 1024 * 1024,
-                             compressCodec: CompressionCodecName = CompressionCodecName.GZIP,
-                             disableDictionaryEncoding: Boolean = false) {
-    log.info("Saving directly as Parquet from SQL. Options other than compression codec are ignored.")
-    dataset.toDF()
-      .write
-      .format("parquet")
-      .option("spark.sql.parquet.compression.codec", compressCodec.toString.toLowerCase())
-      .save(filePath)
-    saveMetadata(filePath)
-  }
-
   override def transformDataset(
     tFn: Dataset[FeatureProduct] => Dataset[FeatureProduct]): FeatureRDD = {
     copy(dataset = tFn(dataset))
   }
 
+  /**
+   * Replaces the sequence dictionary attached to a DatasetBoundFeatureRDD.
+   *
+   * @param newSequences The new sequence dictionary to attach.
+   * @return Returns a new DatasetBoundFeatureRDD with the sequences replaced.
+   */
   def replaceSequences(newSequences: SequenceDictionary): FeatureRDD = {
     copy(sequences = newSequences)
   }
 
+  /**
+   * Replaces the sample metadata attached to the DatasetBoundFeatureRDD.
+   *
+   * @param newSamples The new sample metadata to attach.
+   * @return A DatasetBoundFeatureRDD with new sample metadata.
+   */
+  def replaceSamples(newSamples: Iterable[Sample]): FeatureRDD = {
```

Comment:
This method is defined with scaladoc in `MultisampleGenomicDataset`, could we use `override def` instead?